otx/algorithms/detection/adapters/mmdet/utils/config_utils.py

```diff
@@ -326,3 +327,58 @@ def cluster_anchors(recipe_config: Config, dataset: DatasetEntity):
     config_generator.widths, config_generator.heights = widths, heights
 
     recipe_config.model.bbox_head.anchor_generator = config_generator
+
+
+def patch_tiling(config, hparams, dataset=None):
+    """Update config for tiling.
+
+    Args:
+        config (dict): MPA config containing configuration settings.
+        hparams (DetectionConfig): DetectionConfig containing hyperparameters.
+        dataset (DatasetEntity, optional): A dataset entity. Defaults to None.
+
+    Returns:
+        dict: The updated configuration dictionary.
+    """
+    if hparams.tiling_parameters.enable_tiling:
+        logger.info("Tiling enabled")
+
+        if dataset and dataset.purpose != DatasetPurpose.INFERENCE and hparams.tiling_parameters.enable_adaptive_params:
+            adaptive_tile_params(hparams.tiling_parameters, dataset)
+            tiling_params = ConfigDict(
+                tile_size=int(hparams.tiling_parameters.tile_size),
+                overlap_ratio=float(hparams.tiling_parameters.tile_overlap),
+                max_per_img=int(hparams.tiling_parameters.tile_max_number),
+            )
+            config.update(
```

Comment:
What if hparams.tiling_parameters.enable_adaptive_param is False? Is there another logic to add default tiling_params to config or there is no need to add tiling_params to config in that case?

Reply:
If enable adaptive param is set to false then the default tiling parameters is used:  https://github.com/openvinotoolkit/training_extensions/blob/fd1e6c3cd27c35add8cae76ab474cffd9d5e0c5d/otx/algorithms/detection/configs/detection/configuration.yaml#L411

Reply:
Yes but those default parameters are updated to config? In this code config is updated only when adaptive tiling params is true

Reply:
Tiling parameters already exist in the config as `tile_pipeline.py` is being used instead of using `data_pipeline` https://github.com/openvinotoolkit/training_extensions/blob/5ae47128adc6d312d2b0ae1d29e8e1c0e1a1f5ae/otx/algorithms/common/tasks/training_base.py#L213